### PR TITLE
added action modal for word report modal pop up

### PR DESCRIPTION
**Pull request by bishwobhandari**
- added modal pop up when download report is clicked
- todo for the api url

Implements #162

### CODE REVIEW

File: frontend/src/components/Main/Actions/ActionModal.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React, {Component} from "react";
+import {inject, observer} from "mobx-react";
+import {Modal, Row, Col} from "react-bootstrap";
+import SelectInput from "../../common/SelectInput";
+import {optionType} from "../../../constants/mainConstants";
+import PropTypes from "prop-types";
+
+@inject("mainStore")
+@observer
+class ActionModal extends Component {
```
Comment (shapiromatron) [inline frontend/src/components/Main/Actions/ActionModal.js:10]: rename to WordReportOptionsModal

File: frontend/src/components/Main/Actions/ActionModal.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                onHide={() => mainStore.closeActionModal()}
+                size="m"
+                aria-labelledby="contained-modal-title-vcenter"
+                centered>
+                <Modal.Header>
+                    <Modal.Title id="contained-modal-title-vcenter">Download Report</Modal.Title>
+                    <button
+                        id="close-modal"
+                        className="btn btn-danger"
+                        style={{float: "right"}}
```
Comment (shapiromatron) [inline frontend/src/components/Main/Actions/ActionModal.js:25]: use `float-right` class in bootstrap https://getbootstrap.com/docs/4.0/utilities/float/

File: frontend/src/components/Main/Actions/ActionModal.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                        onClick={() => mainStore.closeActionModal()}>
```
Comment (shapiromatron) [inline frontend/src/components/Main/Actions/ActionModal.js:26]: change to `onClick={mainStore.closeActionModal}`; no need to wrap in a function

File: frontend/src/components/Main/Actions/ActionModal.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                            <button
+                                id="close-modal"
+                                className="btn btn-primary mr-2"
+                                onClick={() => mainStore.downloadReport("wordUrl")}>
+                                Download Report
+                            </button>
+                            <button
+                                id="close-modal"
+                                className="btn btn-danger"
+                                onClick={() => mainStore.closeActionModal()}>
```
Comment (shapiromatron) [inline frontend/src/components/Main/Actions/ActionModal.js:54]: dont wrap this one either

File: frontend/src/components/Main/Actions/ActionModal.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                </Modal.Header>
+                <Modal.Body>
+                    <Row>
+                        <Col>
+                            <SelectInput
+                                label="Select Option"
+                                choices={optionType.map(option => {
+                                    return {value: option.value, text: option.name};
+                                })}
+                                onChange={value => mainStore.updateSelectedModel(value)}
```
Comment (shapiromatron) [inline frontend/src/components/Main/Actions/ActionModal.js:38]: dont wrap this one either

File: frontend/src/stores/MainStore.js [hunk 6]
```diff
@@ -348,6 +354,18 @@ class MainStore {
...
+    @action showActionModal() {
+        this.showModal = true;
+    }
+
+    @action closeActionModal() {
+        this.showModal = false;
+    }
+
+    @action.bound updateSelectedModel(value) {
+        //TODO
```
Comment (shapiromatron) [inline frontend/src/stores/MainStore.js:366]: todo what?

File: frontend/src/stores/MainStore.js [hunk 7]
```diff
@@ -348,6 +354,18 @@ class MainStore {
         this.showToast = false;
     }
     // *** END TOAST ***
+    @action showActionModal() {
```
Comment (shapiromatron) [inline frontend/src/stores/MainStore.js:357]: add `action.bound` for this and below

File: frontend/src/stores/MainStore.js [hunk 8]
```diff
@@ -316,10 +318,14 @@ class MainStore {
     @observable toastHeader = "";
     @observable toastMessage = "";
     @action.bound downloadReport(url) {
+        this.showModal = false;
         let apiUrl = (apiUrl = this.config[url]);
         if (this.canEdit) {
             apiUrl = `${apiUrl}?editKey=${this.config.editSettings.editKey}`;
         }
+        if (url === "wordUrl") {
```
Comment (shapiromatron) [inline frontend/src/stores/MainStore.js:326]: no string w/ +; use the formatting with backticks, this is consistent wit the canEdit line above

File: frontend/src/stores/MainStore.js [hunk 9]
```diff
@@ -21,6 +21,8 @@ class MainStore {
     @observable hasEditSettings = false;
     @observable executionOutputs = null;
     @observable isUpdateComplete = false;
+    @observable showModal = false;
```
Comment (shapiromatron) [inline frontend/src/stores/MainStore.js:24]: showModal is not a good variable name; let's make it more precise `showWordReportOptionModal`

File: frontend/src/components/Main/Actions/Actions.js [hunk 10]
```diff
@@ -89,7 +89,7 @@ class Actions extends Component {
                             <button
                                 className="dropdown-item"
                                 type="button"
-                                onClick={() => mainStore.downloadReport("wordUrl")}>
+                                onClick={() => mainStore.showActionModal()}>
```
Comment (shapiromatron) [inline frontend/src/components/Main/Actions/Actions.js:92]: no func wrapper

**Review by shapiromatron — CHANGES_REQUESTED** [review]
Can't test code, when I tried creating a new session:

```
mobxreact.esm.js?4f20:319 Uncaught TypeError: Cannot read properties of undefined (reading 'map')
    at ActionModal.render (ActionModal.js?24ff:1)
    at allowStateChanges (mobx.module.js?daf9:992)
    at eval (mobxreact.esm.js?4f20:312)
    at trackDerivedFunction (mobx.module.js?daf9:775)
    at Reaction.track (mobx.module.js?daf9:1809)
    at ActionModal.reactiveRender [as render] (mobxreact.esm.js?4f20:310)
    at ActionModal.makeComponentReactive (mobxreact.esm.js?4f20:325)
    at ActionModal.target.render (mobxreact.esm.js?4f20:238)
    at finishClassComponent (react-dom.development.js?61bb:17485)
    at updateClassComponent (react-dom.development.js?61bb:17435)
```

File: frontend/src/components/Main/Actions/ActionModal.js (hunk 1 above)
Comment (bishwobhandari) [reply frontend/src/components/Main/Actions/ActionModal.js:10]: changed

File: frontend/src/stores/MainStore.js (hunk 8 above)
Comment (bishwobhandari) [reply frontend/src/stores/MainStore.js:326]: i replaced string concat to using backticks.

File: frontend/src/stores/MainStore.js (hunk 7 above)
Comment (bishwobhandari) [reply frontend/src/stores/MainStore.js]: done

File: frontend/src/components/Main/Actions/ActionModal.js (hunk 3 above)
Comment (bishwobhandari) [reply frontend/src/components/Main/Actions/ActionModal.js]: if i dont wrap it around, modal pop up does not work.

File: frontend/src/components/Main/Actions/ActionModal.js (hunk 4 above)
Comment (bishwobhandari) [reply frontend/src/components/Main/Actions/ActionModal.js]: same thing, need to wrap for modal pop up to work

Comment (bishwobhandari) [reply frontend/src/components/Main/Actions/ActionModal.js]: we have done same thing in individual ModelDetailModal.js

File: frontend/src/components/Main/Actions/ActionModal.js (hunk 2 above)
Comment (bishwobhandari) [reply frontend/src/components/Main/Actions/ActionModal.js]: done

**Comment by bishwobhandari** [comment]
@shapiromatron i pulled clean state from main this time, and pushed again. for the function wrap in modal. it is required for modal to work. may be for react, it will try to update during rendering and causing problem.
ready for review again, @shapiromatron

**Comment by bishwobhandari** [comment]
i created a sample model option in main constant now for the select options choices, we will update with the models list in second PR.

**Pull request merged**